qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -0,0 +1,857 @@
+
+    Returns:
+        Symbolic equation.
+    """
+    import sympy as sym
+
+    gauss = sym.exp(-(((t - center) / sigma) ** 2) / 2)
+
+    t_edge = zeroed_width / 2
+    offset = sym.exp(-((t_edge / sigma) ** 2) / 2)
+
+    return (gauss - offset) / (1 - offset)
+
+
+def _evaluate_ite(ite_lambda_func: List[Union[Callable, List]], *values: complex) -> bool:
+    """Helper function to evaluate ITE-like symbolic expression.
+
+    Args:
+        ite_lambda_func: Lambdified function which is a list of "IF", "THEN", "ELSE".
+        values: Parameters to be assigned.
+
+    Returns:
+        Result of evaluation.
+
+    Raises:
+        PulseError: When ITE expression is in the wrong syntax.
+    """
+    try:
+        if_func, then_func, else_func = ite_lambda_func
+    except ValueError as ex:
+        raise PulseError(
+            f"Invalid ITE expression with length {len(ite_lambda_func)}. "
+            "This should be a list of callables for evaluating if, then, else."
+        ) from ex
+
+    if bool(if_func(*values)):
+        if isinstance(then_func, list):
+            # Nested if clause
+            return _evaluate_ite(then_func, *values)
+        return bool(then_func(*values))
+    if isinstance(else_func, list):
+        # Nested if clause
+        return _evaluate_ite(else_func, *values)
+    return bool(else_func(*values))
+
+
+def _lambdify_ite(ite_expr: List["Expr"], params: List["Symbol"]) -> List[Union[Callable, List]]:
+    """Helper function to lambdify ITE-like symbolic expression.
+
+    Args:
+        ite_expr: Symbolic expression which is a list of "IF", "THEN", "ELSE".
+        params: Symbols to be used in the expression.
+
+    Returns:
+        Lambdified ITE-like expression.
+    """
+    import sympy as sym
+
+    lambda_lists = []
+    for expr in ite_expr:
+        if isinstance(expr, list):
+            lambda_lists.append(_lambdify_ite(expr, params))
+        else:
+            lambda_lists.append(sym.lambdify(params, expr))
+    return lambda_lists
+
+
+class EnvelopeDescriptor:
+    """Descriptor of pulse envelope.
+
+    When the descriptor of the symbolic pulse subclass is called for the first time,
+    it calls sympy lambdify to create callable and caches the function in the descriptor,
+    which alleviates computational overhead for compiling the symbolic equation.
+    This improves performance of repeated evaluation of waveform samples.
+
+    This descriptor is used when the :class:`SymbolicPulse` creates waveform samples.
+    """
+
+    # Dictionary to store envelope lambda functions for symbolic pulse subclass.
+    # This is keyed on the pulse subclass name.
+    # Note that this is a class variable.
+    # Once envelope is registered here it will be reused for all objects
+    # that will be instantiated in the future, i.e. cache mechanism.
+    global_envelopes = {}
+
+    def __get__(self, instance, owner) -> Callable:
+        clsname = owner.__name__
```

Comment:
Good point. I was thinking to serialize it with class name and create the class on the fly if the name is not found in the import path. However, as you suggested, it will be much cleaner to turn subclass into just a factory and have everything as `SymbolicPulse`.